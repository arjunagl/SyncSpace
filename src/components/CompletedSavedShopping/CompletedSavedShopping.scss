@import "../../stylesheets/styles";
.completedSavedShoppingWrapper {
    display: flex;
    flex-direction: column;
    align-items: center;
    text-align: center;
    @media screen and (min-width: 481px) {
        flex-direction: row;
        align-items: initial;
    }
}

.completedSavedShoppingSeparation {
    width: 100%;
    margin: 0 2rem 0 2rem;
}