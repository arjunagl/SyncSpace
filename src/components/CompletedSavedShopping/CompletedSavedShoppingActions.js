export const CompletedSavedShoppingPathSelected = (shoppingListPath) => ({
    type: 'COMPLETED_SAVED_SHOPPING_LIST_SELECTED',
    AppliedShoppingList: shoppingListPath
});

export const CompletedSavedShoppingPathLoaded = (loadedShoppingListPath) => ({
    type: 'COMPLETED_SAVED_SHOPPING_LIST_LOADED',
    AppliedShoppingList: loadedShoppingListPath
});
