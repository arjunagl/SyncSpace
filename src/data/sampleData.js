export const AppliedShoppingListsSampleData = [
    {
        ShoppingListId: '1',
        Item: {
            Id: '1',
            Name: 'Apples'
        },
        Location: {
            Name: 'Isle 1',
            Description: 'Next to Oranges'
        },
    }, {
        ShoppingListId: '1',
        Item: {
            Id: '2',
            Name: 'Pears'
        },
        Location: {
            Name: 'Isle 1',
            Description: 'Next to Oranges'
        },
    },
    {
        ShoppingListId: '1',
        Item: {
            Id: '3',
            Name: 'Oranges'
        },
        Location: {
            Name: 'Isle 2',
            Description: 'Next to Oranges'
        },
    },
    {
        ShoppingListId: '1',
        Item: {
            Id: '4',
            Name: 'Grapes'
        },
        Location: {
            Name: 'Isle 1',
            Description: 'Next to Oranges'
        },
    },
    {
        ShoppingListId: '1',
        Item: {
            Id: '5',
            Name: 'Carrots'
        },
        Location: {
            Name: 'Isle 4',
            Description: 'Next to Oranges'
        },
    },
    {
        ShoppingListId: '1',
        Item: {
            Id: '6',
            Name: 'Apricots'
        },
        Location: {
            Name: 'Isle 3',
            Description: 'Next to Oranges'
        },
    }
];
