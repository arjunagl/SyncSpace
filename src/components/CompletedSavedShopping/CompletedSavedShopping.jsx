import React from 'react';
import { connect } from 'react-redux';
import commonStyles from '../../stylesheets/styles.scss';
import styles from './CompletedSavedShopping.scss';
import { CompletedSavedShoppingPathSelected } from './CompletedSavedShoppingActions';

const CompletedSavedShoppingPathComponent = ({ CompletedShoppingLists, SavedShoppingLists, onShoppingListPathSelected, history }) => {
    const selectedShoppingList = (completedSavedShoppingList) => {
        history.push('/shopping');
        onShoppingListPathSelected(completedSavedShoppingList);
    };

    //Completed shopping lists
    const completedShoppingPaths = CompletedShoppingLists.map(
        completedShoppingPath =>
            <div
                className={styles.completedSavedShopping__link} key={completedShoppingPath.Name}
                onClick={() => selectedShoppingList(completedShoppingPath)}
            >
                {completedShoppingPath.Name} - {completedShoppingPath.UpdateDate}
            </div>
    );

    //Saved shopping lists
    const savedShoppingPaths = SavedShoppingLists.map(
        savedShoppingPath =>
            <div
                className={styles.completedSavedShopping__link} key={savedShoppingPath.Name}
                onClick={() => selectedShoppingList(savedShoppingPath)}
            >
                {savedShoppingPath.Name} - {savedShoppingPath.UpdateDate}
            </div >
    );
    return (
        <div className={styles.completedSavedShopping__Wrapper}>
            <div className={styles.completedSavedShopping__Separation}>
                <div className={commonStyles.componentWrapper}>
                    <p className={commonStyles.componentTitle}>
                        Completed Shoppings
                    </p>
                    {completedShoppingPaths}
                </div>
            </div>
            <div className={styles.completedSavedShopping__Separation}>
                <div className={commonStyles.componentWrapper}>
                    <p className={commonStyles.componentTitle}>
                        Saved Shoppings
                    </p>
                    {savedShoppingPaths}
                </div>
            </div>
        </div>
    );
};

export { CompletedSavedShoppingPathComponent };

const mapStateToProps = (state) => ({
    CompletedShoppingLists: state.syncSpaceReducer.CompletedShoppingLists,
    SavedShoppingLists: state.syncSpaceReducer.SavedShoppingLists
});

const mapDispatchToProps = (dispatch) => ({
    onShoppingListPathSelected: (shoppingListPath) => {
        dispatch(CompletedSavedShoppingPathSelected(shoppingListPath));
    }
});

const CompletedSavedShoppingPathComponentContainer = connect(mapStateToProps, mapDispatchToProps)(CompletedSavedShoppingPathComponent);
export default CompletedSavedShoppingPathComponentContainer;
