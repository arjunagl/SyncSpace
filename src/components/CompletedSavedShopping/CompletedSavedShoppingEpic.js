import 'rxjs/add/operator/map';
import 'rxjs/add/operator/mergeMap';
import { browserHistory } from 'react-router';
import { CompletedSavedShoppingPathLoaded } from './CompletedSavedShoppingActions';

// eslint-disable-next-line arrow-body-style
export const completedSavedShoppingEpic = (action$, store, { incrementalSearchService }) => {
    // eslint-disable-next-line max-len
    return action$.ofType('COMPLETED_SAVED_SHOPPING_LIST_SELECTED').mergeMap((action) => {
        const loadedShoppingPath = incrementalSearchService.loadSavedCompletedShoppingList(action.AppliedShoppingList);        
        return loadedShoppingPath.map(path => CompletedSavedShoppingPathLoaded(path));
    });
};
