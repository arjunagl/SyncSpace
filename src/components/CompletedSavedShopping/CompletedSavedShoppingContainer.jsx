import React from 'react';
import { connect } from 'react-redux';
import { graphql } from 'react-apollo';
import gql from 'graphql-tag';
import { CompletedSavedShoppingPathComponent } from './CompletedSavedShopping';
import { CompletedSavedShoppingPathSelected } from './CompletedSavedShoppingActions';
import LoaderComponent from '../common/loader/loader';

class CompletedSavedShoppingContainer extends React.Component {
    componentDidMount() {
        //Set the title
        this.props.setTitle({
            windowTitle: 'Completed/Saved Shopping Paths - SyncSpace',
            pageTitle: 'Completed/Saved Shopping Paths'
        });
    }

    render() {
        if (this.props.loading) { //Still loading the data
            return (
                <div style={{ textAlign: 'center' }}>
                    <div style={{ display: 'inline-block' }}>
                        <LoaderComponent />
                    </div>
                </div>
            );
        }
        return (
            <CompletedSavedShoppingPathComponent
                CompletedShoppingLists={this.props.ShoppingPaths.filter(shoppingPath => shoppingPath.completed === true)}
                SavedShoppingLists={this.props.ShoppingPaths.filter(shoppingPath => shoppingPath.completed === false)}
                onShoppingListPathSelected={this.props.onShoppingListPathSelected}
                history={this.props.history}
                loading={this.props.loading}
            />
        );
    }
}

const mapStateToProps = (state) => ({
    CompletedShoppingLists: state.syncSpaceReducer.CompletedShoppingLists,
    SavedShoppingLists: state.syncSpaceReducer.SavedShoppingLists
});

const mapDispatchToProps = (dispatch) => ({
    onShoppingListPathSelected: (shoppingListPath) => {
        dispatch(CompletedSavedShoppingPathSelected(shoppingListPath));
    }
});

const CompletedSavedShoppingPathComponentContainer = connect(mapStateToProps, mapDispatchToProps)(CompletedSavedShoppingContainer);

const getSavedcompletedShoppingPathsQuery = gql`
  query currentUserSavedCompletedShoppingPaths($userId: String!) {
    ShoppingPaths(userId: $userId) {
        Id
        name
        userId
        storeId
        completed
        dateCreated
      }
  }
`;

export default graphql(getSavedcompletedShoppingPathsQuery, {
    options: props => ({ variables: { userId: props.userId } }),
    props: ({ data: { loading, ShoppingPaths } }) => ({
        loading,
        ShoppingPaths,
    }),
})(CompletedSavedShoppingPathComponentContainer);

export const shoppingPathcontainerQuery = getSavedcompletedShoppingPathsQuery;
