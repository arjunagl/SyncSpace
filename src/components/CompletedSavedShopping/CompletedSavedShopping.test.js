import React from 'react';
import { mount, shallow } from 'enzyme';
import sinon from 'sinon';
import toJson from 'enzyme-to-json';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { CompletedSavedShoppingPathComponent } from './CompletedSavedShopping';
import { AppliedShoppingListsSampleData, CompletedShoppingListsSampleData, SavedShoppingListsSampleData } from '../../data/sampleData';
import HistoryMock from '../../../jest/historyMock';

describe('<CompletedSavedShoppingPathComponent />', () => {
    let store;

    beforeEach(() => {
        const mockStore = configureStore();
        const getState = {
            syncSpaceReducer: {
                AppliedShoppingLists: AppliedShoppingListsSampleData
            }
        }; // initial state of the store         
        store = mockStore(getState);
    });

    it('renders the component properly', () => {
        const component = shallow(
            <CompletedSavedShoppingPathComponent
                store={store}
                CompletedShoppingLists={CompletedShoppingListsSampleData}
                SavedShoppingLists={SavedShoppingListsSampleData}
            />
        );
        expect(toJson(component)).toMatchSnapshot();
    });

    it('calls \'onShoppingListPathSelected\' when the user selects a shopping path', () => {
        const onShoppingListPathSelected = sinon.spy();

        const shoppingPathWrapper =
            mount(
                <Provider store={store}>
                    <CompletedSavedShoppingPathComponent
                        CompletedShoppingLists={CompletedShoppingListsSampleData}
                        SavedShoppingLists={SavedShoppingListsSampleData}
                        onShoppingListPathSelected={onShoppingListPathSelected}
                        history={HistoryMock}
                    />
                </Provider>
            );
        const completeShoppingbutton = shoppingPathWrapper.find('[key="CompletedList1"]');
        completeShoppingbutton.simulate('click');
        expect(onShoppingListPathSelected.called).toEqual(true);
    });

    // it('Dispatches the action to the store when the user types in something', () => {
    //     jest.useFakeTimers();
    //     const mockStore = configureStore();
    //     const getState = {}; // initial state of the store 
    //     store = mockStore(getState);

    //     const onPerformIncrementalSearchSpy = sinon.spy();

    //     const incrementalSearchWrapper =
    //         mount(
    //             <IncrementalSearchComponent
    //                 onPerformIncrementalSearch={onPerformIncrementalSearchSpy}
    //                 store={store}
    //             />
    //         );


    //     //find the input element
    //     const searchInput = incrementalSearchWrapper.find('#searchInput');
    //     // searchInput.get(0).value = 'David';
    //     searchInput.simulate('change', { target: { value: 'David' } });
    //     jest.runAllTimers();
    //     const actions = store.getActions();
    //     expect(actions[0]).toEqual(JSON.parse('{"type":"SEARCH_STORES","SearchText":"David"}'));
    // });
});
