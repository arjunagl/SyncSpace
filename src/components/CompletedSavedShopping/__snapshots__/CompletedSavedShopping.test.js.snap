// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CompletedSavedShoppingPathComponent /> renders the component properly 1`] = `
<div>
  <div>
    <div>
      <p>
        Completed Shoppings
      </p>
      <div
        key="CompletedList1"
        onClick={[Function]}
      >
        CompletedList1
         - 
        5/10/2017
      </div>
      <div
        key="CompletedList2"
        onClick={[Function]}
      >
        CompletedList2
         - 
        6/10/2017
      </div>
      <div
        key="CompletedList3"
        onClick={[Function]}
      >
        CompletedList3
         - 
        7/10/2017
      </div>
      <div
        key="CompletedList4"
        onClick={[Function]}
      >
        CompletedList4
         - 
        8/10/2017
      </div>
    </div>
  </div>
  <div>
    <div>
      <p>
        Saved Shoppings
      </p>
      <div
        key="SavedList1"
        onClick={[Function]}
      >
        SavedList1
         - 
        5/8/2017
      </div>
      <div
        key="SavedList2"
        onClick={[Function]}
      >
        SavedList2
         - 
        6/8/2017
      </div>
      <div
        key="SavedList3"
        onClick={[Function]}
      >
        SavedList3
         - 
        7/8/2017
      </div>
      <div
        key="SavedList4"
        onClick={[Function]}
      >
        SavedList4
         - 
        8/8/2017
      </div>
    </div>
  </div>
</div>
`;
